lib/check/check.py

```diff
@@ -215,7 +221,51 @@ def run_check(check, audit_info, output_options):
         logger.error(
             f"{check.checkID} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
         )
-    else:
-        report(findings, output_options, audit_info)
     finally:
-        pass
+        return findings
+
+
+def execute_checks(
+    checks_to_execute: list,
+    provider: str,
+    audit_info: AWS_Audit_Info,
+    audit_output_options: Output_From_Options,
+) -> list:
+    all_findings = []
+    orange = "\033[38;5;208m"
+    print(
+        f"{Style.BRIGHT}Executing {len(checks_to_execute)} checks, please wait...{Style.RESET_ALL}\n"
+    )
+    with alive_bar(
+        total=len(checks_to_execute),
+        ctrl_c=False,
+        bar="blocks",
+        spinner="classic",
+        stats=False,
+        enrich_print=False,
+    ) as bar:
+        for check_name in checks_to_execute:
+            # Recover service from check name
+            service = check_name.split("_")[0]
+            bar.title = f"-> Scanning {orange}{service}{Style.RESET_ALL} service"
+            try:
+                # Import check module
+                check_module_path = (
+                    f"providers.{provider}.services.{service}.{check_name}.{check_name}"
+                )
+                lib = import_check(check_module_path)
+                # Recover functions from check
+                check_to_execute = getattr(lib, check_name)
+                c = check_to_execute()
+                # Run check
+                check_findings = run_check(c, audit_output_options)
+                all_findings.extend(check_findings)
+                report(check_findings, audit_output_options, audit_info)
+                bar()
+
+            # If check does not exists in the provider or is from another provider
+            except ModuleNotFoundError:
```

Comment:
Include another exception handling for a generic Exception

Reply:
Done